packages/node_modules/@webex/internal-plugin-services/src/services.js

```diff
@@ -0,0 +1,227 @@
+import Url from 'url';
+
+import {WebexPlugin} from '@webex/webex-core';
+
+import ServiceUrl from './service-url';
+
+/* eslint-disable no-underscore-dangle */
+/**
+ * @class
+ */
+const Services = WebexPlugin.extend({
+  namespace: 'Services',
+
+  status: {
+    ready: false
+  },
+
+  _serviceUrls: [],
+
+  _preDiscoveryServiceUrls: [
+    new ServiceUrl({
+      name: 'u2c',
+      defaultUrl: process.env.U2C_SERVICE_URL ||
+        'https://u2c.wbx2.com/u2c/api/v1',
+      hosts: []
+    })
+  ],
+
+  /**
+   * Generate a service catalog in the shape of an object.
+   * @returns {object}
+   */
+  services() {
+    const output = {};
+
+    this._serviceUrls.forEach((su) => {
+      output[su.name] = su.get();
+    });
+
+    return output;
+  },
+
+  /**
+   * **Internal**
+   * Search the service url array to locate a `ServiceUrl`
+   * class object based on its name.
+   * @param {string} name
+   * @returns {ServiceUrl}
+   */
+  _getUrl(name) {
+    return this._serviceUrls.find((su) => su.name === name);
+  },
+
+  /**
+   * Get a service url from the current services list by name.
+   * @param {string} name
+   * @returns {string|void}
```

Comment:
I believe `undefined` may work better here, as it reflects the value is `undefined` vs being set to an empty string. I do, however, understand this would make the function return a consistent datatype. Which would be preferred in this case?

Reply:
Let's do `undefined`.